File: .Net/UnitTest/src/Clothe.Application/CreateClothe/ClotheSizeCalculation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Clothes.Application.CreateClothe
{
    public class ClotheSizeCalculation
    {
        public ClotheSizeCalculation() { 
        
        }

        public string Calculate(int height, int width)
        {
            if (height <= 0 || width <= 0)
            {
                throw new InvalidDataException("Invalid height, width");
            }

            if (height <= 40 && width <= 20)
            {
                return "S";
            }

            if (height >= 80 && height  <=120 && width >= 60 && width <=100)
            {
                return "L";
            }

            if (height >= 40 && height  < 80 && width >= 20 && width <60)
            {
                return "M";
            }

            return "XXL";
        }
    }
}

File: Python/Clothes/Clothe/Application/CreateClothe/ClotheSizeCalculation.py
class ClotheSizeCalculation:
    def __init__(self):
        pass

    def calculate(self, height: int, width: int) -> str:
        if height <= 0 or width <= 0:
            raise ValueError("Invalid height, width")

        if height <= 40 and width <= 20:
            return "S"

        if 80 <= height <= 120 and 60 <= width <= 100:
            return "L"

        if 40 <= height < 80 and 20 <= width < 60:
            return "M"

        return "XXL"



